### PR TITLE
IRO-1665 - Added Autocomplete component.

### PR DESCRIPTION
**Pull request by ndemidovich-sc**
Autocomplete

### CODE REVIEW

File: jest.utils.tsx [hunk 1]
```diff
@@ -46,6 +45,5 @@ export const render = (C: any, options?: Omit<RenderOptions, 'wrapper'>) =>
     ...options,
   })
 /* eslint-enable @typescript-eslint/no-explicit-any */
-/* eslint-enable react/display-name */
```
Comment (ndemidovich-sc) [inline jest.utils.tsx:49]: https://github.com/iron-fish/ui-kit/runs/5718711684?check_suite_focus=true

File: components/Autocomplete.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  )
+
+  const optionsToDisplay = options.filter(option => {
+    return filterOption
+      ? filterOption(option, search)
+      : (typeof option?.label === 'string' &&
+          option.label.toLowerCase().includes(search?.toLowerCase())) ||
+          (typeof option?.helperText === 'string' &&
+            option.helperText.toLowerCase().includes(search?.toLowerCase()))
+  })
```
Comment (azhurauski-sc) [inline components/Autocomplete.tsx:87]: Can this be a defaultProps for filterOption function?

**Review by azhurauski-sc — APPROVED** [review]
One minor comment

File: components/Autocomplete.tsx (hunk 2 above)
Comment (brekk) [reply components/Autocomplete.tsx:87]: This function is hard to read. Are the optional chains required? From the type of `OptionType` it doesn't seem like it?

```ts
options.filter((option: OptionType) => {
  const { label: _label, helperText: _helperText } = option
  const label = _label.toLowerCase()
  const help = _helperText.toLowerCase()
  const searchString = search.toLowerCase()
  return filterOption
         ? filterOption(option, search)
         : label && label.includes(search) ||
           help && help.includes(search)
})
```

File: components/Autocomplete.tsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  const { onOpen, onClose, isOpen } = useDisclosure()
+
+  useOutsideClickHandler([inputRef, popoverRef], onClose)
+  useEffect(() => {
+    setVal(value)
+  }, [value])
+
+  const renderDefaultOption = option => (
+    <Box sx={styles?.option}>
+      <Box className={bem('option-label')}>{option?.label}</Box>
```
Comment (brekk) [inline components/Autocomplete.tsx:67]: Will `option?.label` ever be undefined? If so we should be checking for it to exist before trying to render it. If not, we should skip the optional chaining `?.`

File: components/Autocomplete.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+                sx={styles?.optionWrapper}
+                onClick={() => {
+                  if (val !== option) {
+                    setVal(option)
+                    onSelectOption(option)
+                    onClose()
+                  }
+                }}
+              >
+                {(renderOption || renderDefaultOption)(option)}
```
Comment (brekk) [inline components/Autocomplete.tsx:164]: This construct is weird, can we default `renderOption` to `renderDefaultOption` on [47](https://github.com/iron-fish/ui-kit/pull/13/files#diff-1af709b7b4390b50911df9f2d11a10a052d08afa2a527bc35c63933ec7246649R47)?

```
renderOption = renderDefaultOption
```

**Review by brekk — CHANGES_REQUESTED** [review]
Minor suggestions, mainly looking good.

**Review by brekk — APPROVED** [review]

**Pull request merged**